redash/query_runner/qubole.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import absolute_import
+import os
+import json
+import time
+import datetime
+import httplib2
+
+from redash.query_runner import BaseQueryRunner, register
+from redash.query_runner import TYPE_STRING, TYPE_DATE, TYPE_DATETIME, TYPE_INTEGER, TYPE_FLOAT, TYPE_BOOLEAN
+from redash.utils import json_dumps
+
+
+class Qubole(BaseQueryRunner):
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "endpoint": {
+                    "type": "string",
+                    "title": "API Endpoint",
+                    "default": "https://api.qubole.com"
+                },
+                "token": {
+                    "type": "string",
+                    "title": "Auth Token"
+                },
+                "cluster": {
+                    "type": "string",
+                    "title": "Cluster Label",
+                    "default": "default"
+                }
+            },
+            "order": ["endpoint", "token", "cluster"],
+            "required": ["endpoint", "token", "cluster"],
+            "secret": ["token"]
+        }
+
+    @classmethod
+    def enabled(cls):
+        return True
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    def test_connection(self):
+        headers = {"Content-type": "application/json", "Accept": "application/json",
+                   "X-AUTH-TOKEN": self.configuration['token']}
+        h = httplib2.Http()
+        resp, content = h.request("%s/api/latest/users" % self.configuration['endpoint'],
+                                  "HEAD", headers=headers)
+        resp.status == 200
+
+    def run_query(self, query, user):
+        try:
+            import qds_sdk
```

Comment:
Please use a similar pattern to what we have here:
https://github.com/getredash/redash/blob/1b7cc3731c6ad5337855bcadbd5fe1ea64287691/redash/query_runner/big_query.py#L17-L26

This way it will also properly indicate whether the data source type is really supported.

Reply:
done